### PR TITLE
Work flow of order history

### PR DESCRIPTION
**Pull request by vidhushibackbencher**

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -215,30 +219,43 @@ model OrderItem {
...
-  timestamp DateTime @default(now())
-  orderId   Int
-  order     Order    @relation(fields: [orderId], references: [id])
+  id          Int      @id @default(autoincrement())
+  updatedById Int
+  updatedBy    User     @relation(fields: [updatedById], references: [id])
+  timestamp   DateTime @default(now())
+  orderItemId Int
+  statusId    Int  
+
```
Comment (rishiyadav29) [inline prisma/schema.prisma:248]: Remove the extra space here

File: src/order-history/order-history.service.ts [hunk 2]
```diff
@@ -1,39 +1,52 @@
...
+    async create(createOrderHistoryDto: CreateOrderHistoryDto) {
+      
+        const {orderItemId,statusId,updatedById } = createOrderHistoryDto;
 
         const existingOrderHistory = await this.prisma.orderHistory.findFirst({
             where: {
-                orderId: orderId,
-                status: status,
+                orderItemId: orderItemId,
+                statusId:statusId
```
Comment (rishiyadav29) [inline src/order-history/order-history.service.ts:23]: Run lint in this file

File: src/order-items/order-items.service.ts [hunk 3]
```diff
@@ -41,13 +39,70 @@ export class OrderItemsService {
...
+        });
 
-        return await this.prisma.orderItem.findUnique({
-            where: {
-                id: id
+        if (!data) {
+            throw new NotFoundException('Invalid Order Id');
+        }
+
+        const sequence = Array.isArray(data.workflow?.sequence) ? data.workflow.sequence : [];
```
Comment (rishiyadav29) [inline src/order-items/order-items.service.ts:59]: Why is this optional is required here? Verify without the optional and provide the reason for it's use

File: src/order/order.service.ts [hunk 4]
```diff
@@ -28,6 +28,7 @@ export class OrderService {
       }
     }
   
+
```
Comment (rishiyadav29) [inline src/order/order.service.ts:31]: Remove this extra line and run lint

File: src/product/product.service.ts [hunk 5]
```diff
@@ -74,7 +81,13 @@ export class ProductService {
             parentId: true,
           }
         },
-        attributes: true
+        attributes: true,
+        workflow:{
+select:{
```
Comment (rishiyadav29) [inline src/product/product.service.ts:86]: Run lint here

File: src/work-flow/dto/work-flow.create-and-update.dto.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import { IsArray, IsInt, IsNotEmpty, IsOptional, IsString } from "class-validator";
+
+export class CreateWorkFlowDto {
```
Comment (rishiyadav29) [inline src/work-flow/dto/work-flow.create-and-update.dto.ts:3]: Run lint in this

File: src/work-flow/work-flow.controller.ts [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+        this.responseService.sendBadRequest(res, 'Failed to Workflow Delete');
+      }
+    } catch (error) {
+      this.responseService.sendInternalError(res, 'Something Went Wrong');
+    }
+  }
+
+  @Patch(':id')
+  @UseGuards(AuthGuard('jwt'))
+  async updateRole(
```
Comment (rishiyadav29) [inline src/work-flow/work-flow.controller.ts:120]: Why is controller here is named as UpdateRole while you are updating Workflow?

File: src/order-items/order-items.service.ts (hunk 3 above)
Comment (vidhushibackbencher) [reply src/order-items/order-items.service.ts]: I skipped in the mind, why did use this optional?

**Pull request merged**